docs/pages/extensibility/CustomTasks.md

```diff
@@ -106,46 +108,184 @@ A custom task implementation needs to return a function with the following signa
+ * @returns {Promise<Set>}
+ *      Promise resolving with a Set containing all dependencies
+ *      that should be made available to the task.
+ *      UI5 Tooling will ensure that those dependencies have been
+ *      built before executing the task.
+ */
+module.exports.determineRequiredDependencies = async function({availableDependencies, getProject, getDependencies, options}) {
+	// "availableDependencies" could look like this: Set(3) { "sap.ui.core", "sap.m", "my.lib" }
+
+	// Reduce list of required dependencies: Do not require any UI5 framework projects
+	availableDependencies.forEach((depName) => {
+		if (getProject(depName).isFrameworkProject()) {
+		    availableDependencies.delete(depName)
+		}
 	});
-	await Promise.all(markdownResources.map((resource) => {
-		return workspace.write(resource);
+	// => Only resources of project "my.lib" will be available to the task
+	return availableDependencies;
+}
+````
+
+### Examples
+
+The following code snippets show examples for custom task implementations.
+
+### Example: lib/tasks/renderMarkdownFiles.js
+
+````javascript
+const path = require("path");
+const renderMarkdown = require("./renderMarkdown");
+
+/*
+ * Render all .md (Markdown) files in the project to HTML
+ */
+module.exports = async function({workspace, dependencies, log, taskUtil, options}) {
+	const {createResource} = taskUtil.resourceFactory;
+	const textResources = await workspace.byGlob("**/*.md");
+	await Promise.all(textResources.map(async (resource) => {
+		const markdownResourcePath = resource.getPath();
+
+		log.info(`Rendering markdown file ${markdownResourcePath}...`);
+		const htmlString = await renderMarkdown(await resource.getString(), options.configuration);
+
+		// Note: @ui5/fs virtual paths are always (on *all* platforms) POSIX. Therefore using path.posix here
+		const newResourceName = path.posix.basename(markdownResourcePath, ".md") + ".html";
+		const newResourcePath = path.posix.join(path.posix.dirname(markdownResourcePath), newResourceName);
+
+		const markdownResource = createResource({
+			path: newResourcePath,
+			string: htmlString
+		});
+		await workspace.write(markdownResource);
 	}));
 };
 ````
 
 !!! warning
-    Depending on your project setup, UI5 Tooling tends to open many files simultaneously during a build. To prevent errors like `EMFILE: too many open files`, we urge custom task implementations to use the [graceful-fs](https://github.com/isaacs/node-graceful-fs#readme) module as a drop-in replacement for the native `fs` module.
+    Depending on your project setup, UI5 Tooling tends to open many files simultaneously during a build. To prevent errors like `EMFILE: too many open files`, we urge custom task implementations to use the [graceful-fs](https://github.com/isaacs/node-graceful-fs#readme) module as a drop-in replacement for the native `fs` module in case it is used.
+
+    Tasks should ideally use the reader/writer APIs provided by UI5 Tooling for working with project resources.
```

Comment:
are the whitespace in front of the line desired?

Reply:
Yes, that way they appear within a "warning" panel: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage